src/test/java/hw2/ex2/SecondExerciseTest.java

```diff
@@ -0,0 +1,166 @@
+package hw2.ex2;
+
+import hw2.common.AbstractBaseTest;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class SecondExerciseTest extends AbstractBaseTest {
+
+    @Test
+    public void secondExerciseTest() {
+
+        SoftAssert softAssert = new SoftAssert();
+
+        // 1. Open test site by URL
+        driver.get("https://epam.github.io/JDI/");
+
+        // 2. Assert Browser title
+        assertEquals(driver.getTitle(), "Home Page");
+
+        // 3. Perform login
+        driver.findElement(By.id("user-icon")).click();
+        driver.findElement(By.id("name")).sendKeys("epam");
+        driver.findElement(By.id("password")).sendKeys("1234");
+        driver.findElement(By.id("login-button")).click();
+
+        // 4. Assert User name in the left-top side of screen that user is loggined
+        String actualLoggedUser = driver.findElement(By.id("user-name")).getText();
+        assertEquals(actualLoggedUser, "PITER CHAILOVSKII");
+
+        // 5. Click on "Service" subcategory in the header
+        // and check that drop down contains options
+        WebElement serviceHeaderSubMenu = driver.findElement(By.
+                xpath("//ul[@class='uui-navigation nav navbar-nav m-l8']/li[@class='dropdown']"));
+        serviceHeaderSubMenu.click();
+        WebElement serviceSupportSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Support')]"));
+        softAssert.assertTrue(serviceSupportSubCategory.isDisplayed());
+        WebElement serviceDatesSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Dates')]"));
+        WebElement serviceComplTableSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Complex Table')]"));
+        WebElement serviceSimplTableSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Simple Table')]"));
+        WebElement serviceTableWithPagesSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Table with pages')]"));
+        WebElement serviceDiffElemSubCategory =
+                driver.findElement(By.xpath("//ul[@class='dropdown-menu']//li//a[contains(.,'Different elements')]"));
```

Comment:
Is it possible collapse these elements to the loop?

Reply:
fixed